includes/libraries/tracks/class-tracks-client.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+/**
+ * Send Tracks events on behalf of a user.
+ *
+ * @class   Tracks_Client
+ * @package WooCommerce/Classes
+ *
+ *
+ * Example Usage:
+```php
+	require( dirname(__FILE__).'path/to/tracks/class-tracks-client.php' );
+
+	$result = Tracks_Client::record_event( array(
+		'_en'        => $event_name,       // required
+		'_ui'        => $user_id,          // required unless _ul is provided
+		'_ul'        => $user_login,       // required unless _ui is provided
+
+		// Optional, but recommended
+		'_ts'        => $ts_in_ms,         // Default: now
+		'_via_ip'    => $client_ip,        // we use it for geo, etc.
+
+		// Possibly useful to set some context for the event
+		'_via_ua'    => $client_user_agent,
+		'_via_url'   => $client_url,
+		'_via_ref'   => $client_referrer,
+	) );
+
+	if ( is_wp_error( $result ) ) {
+		// Handle the error in your app
+	}
+```
+ */
+
+require_once dirname( __FILE__ ) . '/class-tracks-event.php';
+
+/**
+ * Class Tracks_Client.
+ */
+class Tracks_Client {
+	const PIXEL           = 'https://pixel.wp.com/t.gif';
+	const BROWSER_TYPE    = 'php-agent';
+	const USER_AGENT_SLUG = 'tracks-client';
+
+	/**
+	 * Record a Tracks event
+	 *
+	 * @param  mixed $event Event object to send to Tracks. An array will be cast to object. Required.
```

Comment:
This sound likes `Tracks_Event|array` and is better instead of `mixed`, I don't think that should work with bool, strings and so on.
There's a lot of docblocks in this classes that doesn't describe any param.